### DISCUSSION TITLE
SMTP is not working after updating hubs client

### DISCUSSION THREAD

**Discussion by big-javascript-fan** [mozilla/hubs#3657] [Hubs Cloud Questions] [answered]
I have a project which was installed on AWS on July, 2020.
I made some modification on hubs client in UI features and updated it on October and November.
For now, it doesn't work to send magic link emails at all.

I have another hubs instance on AWS with same region and it send emails so I am sure this is not limit problem on Amazon SES.
This is very serious issue because I am not able to login to admin panel and not update any more.
Project is online with this URL:
https://project7933.com

Are there any issues between hubs client version after October and hubs backend version before June 2020?
Or any other possible issue on email not working?

**Comment by robinkwilson** [comment]
Sounds like your custom hubs client needs to be updated to the latest!

You'll need to merge new changes from the **hubs-cloud** branch in the hubs repo into your custom client, then deploy those updates to your HC instance -- _remember to do this regularly!_

You can find the latest on the hubs-cloud branch in the Hubs github repo. We updated instructions for how to setup an upstream git repo and merge the hubs-cloud branch into your custom client in the [Custom Client documentation](https://hubs.mozilla.com/docs/hubs-cloud-custom-clients.html).

To see when Hubs Cloud was automatically updated, check out the Changelog here: https://github.com/mozilla/hubs-cloud/blob/master/CHANGELOG.md

**Reply by big-javascript-fan** [reply]
Thanks for your reply, but that is not possible.
As I can not get magic link email, I can not make deployment to hubs client.
That's why I can not make it upgrade or downgrade too.

Are there any other way to solve this?

**Reply by robinkwilson** [reply]
To remove the custom client without logging in:

1. Delete the Parameter Store config
     1. Go to AWS Console > Systems Manager > Parameter Store
     1. Delete config called `ita/<your HC stack name>/hubs/deploy/type`
1. Force a config refresh by updating the stack
     1. Go to AWS Console > Cloudformation > Stacks - _check your region! if you see 0 stacks_
     1. Update your EC2 server size from `t3.small` -> `t3.medium` - _you'll change this back after!!!_
1. Check that your HC landing page to see if everything worked! 
     1. In your case see if you can login
1. Change the EC2 server size back to the old config `t3.medium` -> `t3.small`

Let me know if this works for you.

**Reply by Count-On-AG** [reply]
Hi, Robin.  I'm working with Piao on this issue.  Thanks so much for your assistance.  We've come a long way since you and Greg Fodor helped me launch my first Hubs Cloud Instance almost a year ago!  But now we're coming up against a brick wall.  We made some changes to our Hubs Cloud code and everything was working great until it looks like we missed an update.  Piao has tried all of the suggestions above and we're still unable to send or receive the magic link to log into admin panel.  If you have any other suggestions, or resources you could point us towards, we would sure appreciate it.

**Reply by robinkwilson** [reply]
Let's look at region us-east-1 to see if SES is sending you emails (regardless of deploy region, all emails go through N. Virginia (us-east-1) I believe you're in Oregon IIRC)

AWS console > SES menu > in upper right corner check that your region is us-east-1

Quick checks for email issues:
https://hubs.mozilla.com/docs/hubs-cloud-aws-troubleshooting.html#then-what-is-my-hub-stacks-admin-email-address

Next check inside AWS SES (since this is AWS, there are more guides online if you web search if you get confused): 
See if/what emails have been sending. 
Next check that your admin email is verified with AWS. If it is not, you'll need to send the resend the email to verify your admin email.
Next can you check Oregon region that your admin email is also verified?

Also, are you out of the SES sandbox in us-east-1? - _likely on the other deploy you have_

Let me know how this goes glad you're onboard!

**Reply by robinkwilson** [reply]
In your hubs, also make sure there are no failed requests in your developer console, when you request the magic link.

**Comment by robinkwilson** [comment]
Any update on this one? (it was hidden by a "view more" button for the conversation.

**Comment by robinkwilson** [comment, marked as answer]
@big-javascript-fan and I figured out we had [custom SMTP settings](https://hubs.mozilla.com/docs/hubs-cloud-custom-smtp.html) that were still configured and weren't sending emails. 

We were able to get emails back to the AWS defaults by deleting the 4 `email` settings parameters in the Parameter Store (AWS Console > System Settings > Parameter Store).

To ensure the system saw the change, we restarted the instances by "Terminating" the EC2 instance. Quickly a new one came up and emails were working!

**Comment by Count-On-AG** [comment]
Robin:

Thank you SO MUCH for help with this.  As I’ve mentioned earlier, we are so excited to be working with the Hubs platform and see great possibilities for the future!!

You guys are the best.

-Andrew

**Comment by tranvu14** [comment]
I have clone this repo 3 days ago, and deploy my custom client. But I do not receive any email with a new account sign up. This work fine when I use a registered email to sign in, but I still can not sign up. When I reset my stack, it worked. But when I deploy again, It is not working. Anyone face this problem?
